### ISSUE TITLE
Prompted to reload window every time on devcontainer open

### ISSUE THREAD

**Issue by jsma** [microsoft/pylance-release#2529]
## Environment data

-   Language Server version: v2022.3.4
-   OS and version: macOS 12.3.1
-   Python version (& distribution if applicable, e.g. Anaconda): Python 3.8 installed via docker-compose in a dev container

## Expected behaviour

Having zero knowledge of how VS Code internals are supposed to work, as an end user I want to call ``devcontainer build && devcontainer open`` and have VS Code open my project in a dev container and let me get to work.

## Actual behaviour

After the ``devcontainer open``, I have to wait for it to rebuild a bunch of stuff which hasn't changed from the last time I did ``devcontainer build`` (a separate issue) and then after everything is satisfactorily built, I get a prompt: "Pylance extension is now installed. Reload window to activate?"

<img width="578" alt="Screen Shot 2022-04-01 at 2 18 25 PM" src="https://user-images.githubusercontent.com/24574/161342551-23ce2213-2270-4793-821f-a21f93c9145a.png">

I realize there may be technical limitations to how all of this complexity is wired up in the background, but it seems odd to stare at a blank VS Code window which won't let me even see the files in my project while it rebuilds stuff, hoping I can finally start editing files, only to be told I have to reopen the window one more time.

Is it not possible to ``devcontainer open`` after a ``devcontainer build`` and have everything ready to go instead of waiting for an additional build process followed by a reload window operation?

**Comment by thekaveman** [comment]
I see the same notification every time I open (or reopen) my devcontainer, with:

* Language Server version: v2022.4.1
* OS and version: Windows 10 (19043.1645)
* Python version (& distribution if applicable, e.g. Anaconda): `python:3.9-slim` installed via Docker

Often times, clicking "Yes" causes the devcontainer to fail to restart/reopen. Where I have to go delete the existing Docker container and restart from scratch.

I usually click the "X" to close the notification -- sometimes Pylance works, sometimes not.

**Comment by thclark** [comment]
Yeah this is particularly frustrating because it shows up *during* the post-create phase, so if you click 'yes' that breaks anything happening in post-create.

**Comment by bschnurr** [comment]
r you using a dockerfile with the pylance extension listed?
ie. https://code.visualstudio.com/docs/remote/create-dev-container#_dockerfile

```
{
  "build": { "dockerfile": "Dockerfile" },

  "customizations": {
    "vscode": {
      "extensions": [""]
    }
  },

  "forwardPorts": [3000]
}
```

please provide some more info around your devcontainer setup

**Comment by thekaveman** [comment]
From #2723:

> Links:
> 
> * Public repo: https://github.com/cal-itp/benefits
> * [devcontainer.json](https://github.com/cal-itp/benefits/blob/dev/.devcontainer/devcontainer.json#L37)
> * [settings.json](https://github.com/cal-itp/benefits/blob/dev/.vscode/settings.json#L27)
> 
> VS Code / runtime info:
> 
> ```
> Version: 1.67.2 (user setup)
> Commit: c3511e6c69bb39013c4a4b7b9566ec1ca73fc4d5
> Date: 2022-05-17T18:15:52.058Z
> Electron: 17.4.1
> Chromium: 98.0.4758.141
> Node.js: 16.13.0
> V8: 9.8.177.13-electron.0
> OS: Windows_NT x64 10.0.19044
> Language Server version: v2022.5.2 (experienced with v2022.4.1 as well)
> Python version (& distribution if applicable, e.g. Anaconda): python:3.9-slim (Debian GNU/Linux 11) installed via Docker
> ```
> 
> Example in VS Code devcontainer:
> 
> * _Loading..._ instead of `os` docstrings - this never "loads"
> * Lack of highlighting on `os` module/functions
> 
> ![image](https://user-images.githubusercontent.com/1783439/170155826-9fa0b459-bc21-4402-b3f8-eab1f851bfcc.png)

Sometimes (not every time) if I wait for the devcontainer to fully initialize, then do `Developer: Reload Window`, then wait for everything to reload -- _then_ Pylance will work.

**Comment by thekaveman** [comment]
Some new information, definitely backs up @thclark's idea that the sequencing of when Pylance is installed vs. when it is expected to run (or be running) is at issue.

TLDR; Pylance consistently fails in a fresh build / rebuild of the devcontainer. After a rebuild, having opened the devcontainer once already (where Pylance failed), Pylance consistently works when the devcontainer is reopened.

We simplified our devcontainer (build time) considerably if anyone wants to play along:

* Repo: https://github.com/cal-itp/benefits
* Steps to get the devcontainer running: https://docs.calitp.org/benefits/development/

## Repro steps

1. `Rebuild and Reopen in Container` from host machine:
    * See *The Pylance extension is not installed...* warning message
    * No Pylance features are working (syntax highlighting, intellisense, etc.)
    * The language server is seemingly not started, `Output > Python Language Server` remains empty
    * `Restart Language Server` produces output, but does not activate Pylance features
  
2.  After step 1, go back to the host machine with `Reopen Folder Locally`, and then `Reopen in Container`
    * Do not see any warning messages
    * All Pylance features are working (syntax highlighting, intellisense, etc.)
    * `Output > Python Language Server` shows log info almost immediately after start up
    * `Restart Language Server` works as expected, all Pylance features remain working

**Comment by debonte** [comment]
Closing in the belief that this is a vscode issue. If vscode is unable to act on https://github.com/microsoft/vscode/issues/152263, we'll reactivate this issue.

**Comment by debonte** [comment]
Canonical bug is https://github.com/microsoft/vscode/issues/149309

**Comment by oscarhermoso** [comment]
> Closing in the belief that this is a vscode issue. If vscode is unable to act on [microsoft/vscode#152263](https://github.com/microsoft/vscode/issues/152263), we'll reactivate this issue.

Can this be re-opened? Linked issues suggest it was fixed with 1.72.*, but I'm still having the issue on 1.75.1

**Comment by debonte** [comment]
@oscarhermoso, this is unlikely to be a Pylance-specific issue. Please file an issue at https://github.com/microsoft/vscode/issues.
